04/SpeedTest.js

```diff
@@ -1,17 +1,42 @@
-import React, { useEffect } from 'react';
+import React, { useEffect,useState,useRef } from 'react';
 import useRandomItem from './hook';
 
 const SpeedTest = () => {
     const [word, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JavaScript']);
+    const [inputValue, setInputValue] = useState('');
+    const [counter, setCounter] = useState(0);
+    const [charactersLength, setCharactersLength] = useState(0);
+    const inputRef = useRef(null);
+    const intervalRef = useRef(null);
+
+    const stopInterval = () => {
+        clearInterval( intervalRef.current );
+    }
 
     useEffect(() => {
         regenerateWord();
-    }, []);
+        return () => stopInterval();
+    },[]);
+
+    const startGame = () => {
+        intervalRef.current = setInterval(() => {
+            setCounter(value => value + 1);
+        }, 1000);
+    };
+
+    useEffect(() => {
+        if (inputValue === word) {
+            setCharactersLength(value => value + inputValue.length);
+            setInputValue('')
```

Comment:
Czy tutaj nie brakuje również `regenerateWord`? Obecnie ciągle jest to samo słowo do wpisania ;P